hardware_integration/CMakeLists.txt

```diff
@@ -304,6 +308,46 @@ if("SYS_TEC" IN_LIST CAN_DRIVER)
     )
   endif()
 endif()
+if("NTCAN" IN_LIST CAN_DRIVER)
+  if(MSVC)
+    # See https://gitlab.kitware.com/cmake/cmake/-/issues/15170
+    set(CMAKE_SYSTEM_PROCESSOR ${MSVC_CXX_ARCHITECTURE_ID})
+  endif()
+  if(WIN32)
+    if(DEFINED ENV{CanSdkDir})
+      if(CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64" OR CMAKE_SYSTEM_PROCESSOR
+                                                    STREQUAL "x64")
+        message(STATUS "Detected x64, linking to NTCAN x64 library")
+        target_link_libraries(HardwareIntegration
+                              PRIVATE $ENV{CanSdkDir}/lib/VC/amd64/ntcan.lib)
+        target_include_directories(HardwareIntegration
+                                   PUBLIC $ENV{CanSdkDir}/include)
+      elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_SYSTEM_PROCESSOR
+                                                         STREQUAL "X86")
+        message(STATUS "Detected x86, linking to NTCAN x86 library")
+        target_link_libraries(HardwareIntegration
+                              PRIVATE $ENV{CanSdkDir}/lib/VC/i386/ntcan.lib)
+        target_include_directories(HardwareIntegration
+                                   PUBLIC $ENV{CanSdkDir}/include)
+      else()
+        message(
+          FATAL_ERROR
+            "NTCAN Selected but no supported processor arch was detected. Only x64 and x86 are supported."
+        )
+      endif()
```

Comment:
@ad3154 have a look at this part. It dynamically compiles the needed header and library files into the stack based on a system variable already set by NTCan. I think it's a pretty neat approach that doesn't involve any licenses